src/nonunity/actions/NEML2Action.C

```diff
@@ -0,0 +1,142 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifdef NEML2_ENABLED
+
+#include "neml2/misc/utils.h"
+#include "neml2/misc/parser_utils.h"
+#include "neml2/base/HITParser.h"
+#include "NEML2Action.h"
+#include "NEML2Utils.h"
+#include "FEProblem.h"
+#include "Factory.h"
+
+registerMooseAction("BlackBearApp", NEML2Action, "parse_neml2");
+registerMooseAction("BlackBearApp", NEML2Action, "add_material");
+registerMooseAction("BlackBearApp", NEML2Action, "add_user_object");
+
+InputParameters
+NEML2Action::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Action to parse and setup NEML2 objects.");
+  params.addRequiredParam<FileName>("input",
+                                    "Path to the NEML2 input file containing the NEML2 model(s)");
+  params.addRequiredParam<std::string>(
```

Comment:
As a future task we might want to look into creating a custom derived string class and then enable extending the language server to get auto-completion for those valid items.

Reply:
I agree.